### ISSUE TITLE
spotify-rise starts Spotify but it doesn't play music

### ISSUE THREAD

**Issue by j86989** [gcushen/spotify-rise#1]
```
~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04 LTS
Release:    16.04
Codename:   xenial

~$ spotify-rise
Running at 23-06-2016 22:28:35...
Wakey! Wakey! Good morning :D
Cannot find 'org.freedesktop.DBus.Properties.Get' in object /org/mpris/MediaPlayer2 at org.mpris.MediaPlayer2.spotify
```

**Comment by gcushen** [comment]
I am using Ubuntu 16.04 with the latest Spotify 1.0.28.89 and Spotify Rise without any DBus issue.

Please can you provide some more debug info...

Are you using Kubuntu or KDE desktop?

Which version of Spotify are you running (click your username > settings > scroll down > about)?

Do you have the latest Spotify Rise version?

When Spotify is playing a song, what's the output of:

```
qdbus | grep -i spotify
```

**Comment by j86989** [comment]
No, I use Canonical's Ubuntu with Unity.

![screenshot from 2016-06-24 10-03-09](https://cloud.githubusercontent.com/assets/20116585/16331820/f164a798-39f2-11e6-90df-e3438e20d4cf.png)

I have the latest Spotify Rise version.

```
~$ qdbus | grep -i spotify
org.mpris.MediaPlayer2.spotify
```

---

The undermentioned commands work properly:

```
~$ qdbus org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Pause
~$ qdbus org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Play
```

The first one pauses Spotify and the second one plays music.
When line 77:

```
metadata=`qdbus org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.freedesktop.DBus.Properties.Get org.mpris.MediaPlayer2.Player Metadata`
```

is removed, there is no `Cannot find 'org.freedesktop.DBus.Properties.Get' in object /org/mpris/MediaPlayer2 at org.mpris.MediaPlayer2.spotify` message but it still doesn't work.

---

![screenshot from 2016-06-26 00-29-56](https://cloud.githubusercontent.com/assets/20116585/16359463/5a99ddda-3b35-11e6-8d65-240ab30ddef4.png)

**Comment by gcushen** [comment]
I see the issue now. 

The following warning can safely be ignored or redirected to `/dev/null` as the relevant code is just using _metadata_ to check for when Spotify has finished starting up:

```
Cannot find 'org.freedesktop.DBus.Properties.Get' in object /org/mpris/MediaPlayer2 at org.mpris.MediaPlayer2.spotify
```

I use shuffle mode with Spotify 1.0.28.89 and it works fine when given the URI of a track within an album:

```
spotify-rise -s -u spotify:track:41Uz23jubhxB8YREUHvueV
```

However, it appears normal play mode (no `-s`) is not working in conjunction with Spotify 1.0.28.89 - which I suspect may be the mode you were attempting to use. In that case, downgrading to Spotify 0.9 might work better for the reasons stated at the end of the Spotify Rise blog post about the DBus changes between Spotify v0.9 and v1: https://georgecushen.com/spotify-alarm-clock-ubuntu-linux

**Comment by gcushen** [comment]
I have just released v0.3.0 of the script. It addresses this issue by offering much improved support for Spotify v1 as well as general improvements and updated documentation at https://github.com/gcushen/spotify-rise/blob/master/README.md .

Since Spotify **v1** does not support playing playlists/albums via Dbus, you must provide the _URI of a track_ (such as within an album). With shuffle mode active, rather than playing the specified track, it will randomly play a song from the track's parent album. Thus, partially working around the limitations of the Spotify **v1** Dbus implementation. Otherwise, for playlist support, downgrading to Spotify **v0.9** and specifying a _playlist URI_ should work.
